src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,164 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> start;
+    private Node<T> end;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        addToTail(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkBounds(index);
+        if (index == size) {
+            addToTail(value);
+        } else {
+            addWithShift(value, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (final T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return findNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = findNode(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = findNode(index);
+        if (node.prev == null && node.next == null) {
+            start = null;
+            end = null;
+        } else if (node.prev == null) {
+            start = node.next;
+            start.prev = null;
+        } else if (node.next == null) {
+            end = node.prev;
+            end.next = null;
+        } else {
+            node.next.prev = node.prev;
+            node.prev.next = node.next;
+        }
+        size--;
+        return node.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = start;
+        for (int i = 0; i < size; i++) {
+            if ((object == null && object == node.value) || node.value.equals(object)) {
```

Comment:
You will get NPE if **object != null** and **node.value == null**
just use null-safe equals expression here.

Reply:
if I delete (object == null && object == node.value) test testRemoveByItemPositive() is not passed
if you talking about Objects class - we can't use it in this task